drivers/bus/pci/pdo.c

```diff
@@ -1334,15 +1335,19 @@ PdoStartDevice(
         return STATUS_SUCCESS;
 
     /* TODO: Assign the other resources we get to the card */
-
+;
+    RawFullDesc = &RawResList->List[0];
     for (i = 0; i < RawResList->Count; i++)
     {
-        RawFullDesc = &RawResList->List[i];
-
-        for (ii = 0; ii < RawFullDesc->PartialResourceList.Count; ii++)
+        PartialResourceList = &RawFullDesc->PartialResourceList;
+        RawFullDesc = (PCM_FULL_RESOURCE_DESCRIPTOR)
+            (&PartialResourceList->PartialDescriptors[PartialResourceList->Count]);
```

Comment:
Yeah, you're jumping to the second descriptor, without parsing the first one. Or am I missing something?

Reply:
What happens if the list contains a variable length (with DeviceSpecificData) descriptor?
This code always works:
```c
VOID NTAPI ParsingCmResourceList(In PCM_RESOURCE_LIST CmResource)
{
    PCM_FULL_RESOURCE_DESCRIPTOR FullList;
    PCM_PARTIAL_RESOURCE_DESCRIPTOR Descriptor;
    ULONG ix;
    ULONG jx;

    if (!CmResource) {
        return;
    }
    if (!CmResource->Count) {
        return;
    }
    
    FullList = &CmResource->List[0];
    
    for (ix = 0; ix < CmResource->Count; ix++)
    {
        Descriptor = FullList->PartialResourceList.PartialDescriptors;
    
        for (jx = 0; jx < FullList->PartialResourceList.Count; jx++)
        {
            /*
                parsing ...
            */
    
            Descriptor = NextPartialDescriptor(Descriptor);
        }
    
        FullList = (PCM_FULL_RESOURCE_DESCRIPTOR)Descriptor;
    }
}
```
Also, if you are sure that no more than 1 list, then you can shorten one cycle.

Reply:
Yes, and in fact it's even simpler, because there can only be one variable sized descriptor in a partial resource list, and it must be the last one. I updated the code accordingly.